### PR TITLE
feat(rid): Prompt user to change template scope to org

### PR DESCRIPTION
**Pull request by jmtaber129**
# Description

Fixes https://github.com/ParabolInc/parabol/issues/8026

When a template is team-scoped, the activity can only be started on the owning team, so the team dropdown in the details view would only show the owning team.

Instead, show a popover that prompts the user to change the scope to "org".

This loosens the template scope update mutation such that any team member can update the scope (not just team leads)

## Demo
https://www.loom.com/share/0f4486618b4344a2a5924b281befb361

## Testing scenarios
- [ ] Team-scoped template:
  - [ ] Clicking the team selector shows the popover
  - [ ] Clicking the button in the popover updates the scope, and replaces the popover with the team selector menu for teams in the org.
- [ ] Org-scoped templates always show the normal team dropdown
- [ ] Non-team-leads can change template scope.

## Final checklist

- [X] I checked the [code review guidelines](../docs/codeReview.md)
- [X] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [X] I have performed a self-review of my code, the same way I'd do it for any other team member
- [X] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [X] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [X] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [X] PR title is human readable and could be used in changelog

### CODE REVIEW

File: packages/client/components/ActivityLibrary/ActivityDetailsSidebar.tsx [hunk 1]
```diff
@@ -93,6 +99,37 @@ const ActivityDetailsSidebar = (props: Props) => {
...
+
+  const teamScopePopover = templateTeam && selectedTemplate.scope === 'TEAM' && (
+    <div className='w-[352px] p-4'>
+      <div>
+        This custom activity is private to the <b>{templateTeam.name}</b> team.
+      </div>
+      <br />
+      <div>
+        As a member of the team you can share this activity with the{' '}
+        <b>{templateTeam.organization.name}</b> organization so that any team can also use the
```
Comment (mattkrick) [inline packages/client/components/ActivityLibrary/ActivityDetailsSidebar.tsx:118]: +1 nit on the word "any" and below on "all", which some less technically-inclined folks may interpret to mean "all teams [in the app]".
Maybe "allow teams in the X organization..."?

Comment (jmtaber129) [reply packages/client/components/ActivityLibrary/ActivityDetailsSidebar.tsx:118]: Hmm, I could go either way.

@ackernaut this is the copy from your designs, WDYT?

Comment (ackernaut) [reply packages/client/components/ActivityLibrary/ActivityDetailsSidebar.tsx:118]: What do you think about this [iteration on copy in Notion](https://www.figma.com/file/PmE9A2ryODVzFPjZFJn7qi/Parabol-Activity-Library?type=design&node-id=1381%3A32031&t=ChKuSk6LsUmoIzoT-1)

Comment (jmtaber129) [reply packages/client/components/ActivityLibrary/ActivityDetailsSidebar.tsx:131]: @ackernaut that copy LGTM! I'd just change "can use it" to "can use the activity" to be specific, similar to the previous copy.

**Comment by jmtaber129** [comment]
New copy:
Screen Shot 2023-05-04 at 1 55 10 PM

**Pull request merged**